pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -292,6 +292,19 @@ func (c *Reconciler) reconcile(ctx context.Context, pr *v1beta1.PipelineRun) err
 		return nil
 	}
 
+	// build DAG with a list of final tasks, this DAG is used later to identify
```

Comment:
question: i didnt see any update to any validation logic, just want to check that there is something before this that will be invoked to validate the parts fo the finally tasks specific to them (e.g. not allowing "from" with resources) - since we can't always trust that the webhook got called :S

Reply:
yup, `pipeline_validation.go` works as a gatekeeper and [validates](https://github.com/tektoncd/pipeline/blob/648cff3dfe1330c5035492437adee6bf1670d941/pkg/apis/pipeline/v1beta1/pipeline_validation.go#L447) final tasks before reaching here.